src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,51 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> tail;
+    private int size;
 
+    public MateStack() {
+        size = 0;
+        tail = new Node<>(null,null);
```

Comment:
Your stack will always have some extra node which does not contain any value. You can dismiss initializing `tail` because if it is not initialized, it will be null by defaulf (line 6).